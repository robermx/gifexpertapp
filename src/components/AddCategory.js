import React, { useState } from 'react';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handelSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length >= 2) {
      setCategories((cat) => [inputValue, ...cat]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handelSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};
