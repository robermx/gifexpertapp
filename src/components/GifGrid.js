// import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="text-light animate__animated animate__fadeInDown">
        {category}
      </h3>
      {loading && (
        <p className="center animate__animated animate__pulse">Cargando</p>
      )}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
