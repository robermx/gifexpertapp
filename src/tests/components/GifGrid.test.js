import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Puebas con GifGrid component', () => {
  const category = 'lost';

  test('Mostrarse correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Mostrando items cuando se cargan con UseFetchGifs', () => {
    const gifs = [
      {
        id: '1626ty',
        url: 'https://localhost/bacha.jpg',
        title: 'bachas',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
