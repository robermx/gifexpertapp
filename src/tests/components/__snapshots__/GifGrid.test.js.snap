// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas con GifGrid component Mostrando items cuando se cargan con UseFetchGifs 1`] = `
<Fragment>
  <h3
    className="text-light animate__animated animate__fadeInDown"
  >
    lost
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1626ty"
      key="1626ty"
      title="bachas"
      url="https://localhost/bacha.jpg"
    />
  </div>
</Fragment>
`;

exports[`Puebas con GifGrid component Mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="text-light animate__animated animate__fadeInDown"
  >
    lost
  </h3>
  <p
    className="center animate__animated animate__pulse"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
